pkg/loader/loader.go

```diff
@@ -16,87 +16,82 @@ limitations under the License.
 
 package loader
 
-import (
-	"fmt"
+import "fmt"
 
-	"k8s.io/kubectl/pkg/kinflate/util/fs"
-)
-
-// Loader abstracts how bytes are read for manifest, resource, patch, or other
-// files. Each Loader is tightly coupled with the location of a manifest file.
-// So each file without an absolute path is located relative to the
-// manifest file it was read from. Each Load() call is relative to this manifest
-// location (referenced as root). The Loader hides how to read bytes from different
-// "schemes" (e.g. file, url, or git).
+// Loader interface exposes methods to read bytes in a scheme-agnostic manner.
+// The Loader encapsulating a root location to calculate where to read from.
 type Loader interface {
-	// Clones new Loader for a new app/package with its own manifest from the current
-	// Loader. The "newRoot" can be relative or absolute. If it's relative, the new
-	// Loader root is calculated from the current Loader root. Can be a file or directory.
-	// If it's a file, then the base directory is used for root calculation.
+	// Root returns the scheme-specific string representing the root location for this Loader.
+	Root() string
+	// New returns Loader located at newRoot.
 	New(newRoot string) (Loader, error)
-	// Returns the bytes at location or an error. If it's a relative path, then
-	// the location is expanded using the Loader root.
-	// Example: returns YAML bytes at location "/home/seans/project/service.yaml".
+	// Load returns the bytes read from the location or an error.
 	Load(location string) ([]byte, error)
 }
 
 // Private implmentation of Loader interface.
 type loaderImpl struct {
-	root string
-	fs   fs.FileSystem
-	// http client for URL loading
-	// git client for Git loading
+	root    string
+	schemes []SchemeLoader
+}
+
+// Interface for different types of loaders (e.g. fileLoader, httpLoader, etc.)
+type SchemeLoader interface {
+	// Does this location correspond to this scheme.
+	IsScheme(root string, location string) bool
+	// Calulcate a new root.
+	Root(root string, location string) (string, error)
+	// Combines the root and path into a full location string.
+	FullLocation(root string, path string) (string, error)
```

Comment:
> This documentation piece is not printed, because it's documenting a private struct. It's too bad because the exposed documentation is not very clear.

I really don't understand what the difference is between `Root` and `FullLocation` by reading these two comments.